### PR TITLE
Interaction System Documentation

### PR DESCRIPTION
**Pull request by TaylorNAlbarnaz**
## Summary

Documents the remaining classes, interfaces and methods that weren't properly documented

## Changes to Files

Changes pretty much every script in the SS3D/Interactions folder, only additions were comments, and a single unused "InteractionInvalid" event was removed

### CODE REVIEW

**Comment by cosmiccoincidence** [comment]
~~Please link the documentation in the PR description.~~

Nevermind, I see that this is for in-code documentation.

File: Assets/Scripts/SS3D/Interactions/InteractionTargetNetworkBehaviour.cs [hunk 1]
```diff
@@ -4,9 +4,12 @@
 
 namespace SS3D.Interactions
 {
+    /// <summary>
+    /// A base class for interaction targets which use the mirror NetworkBehaviour
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Interactions/InteractionTargetNetworkBehaviour.cs]: correct that to fishnet

**Review by stilnat — DISMISSED** [review]
just a small correction and good to go

**Review by cosmiccoincidence — APPROVED** [review]
Looks good.

**Pull request merged**